### PR TITLE
feat: [modal migration] updated design for new source modal

### PR DESCRIPTION
**Pull request by el**
## Changes

Updated the design of the new source modal according to the new design:
https://www.sketch.com/s/2867ec63-0966-4647-b1d7-b6fe696ba251/a/m1913YJ

### CODE REVIEW

File: packages/shared/src/components/modals/NewSourceModal.tsx [hunk 1]
```diff
@@ -184,8 +183,10 @@ export default function NewSourceModal(props: ModalProps): ReactElement {
           onSubmit={onScrapeSubmit}
           aria-busy={isScraping}
           data-testid={`login state: ${loginState?.trigger}`}
+          id="submit-source"
```
Comment (rebelchris) [inline packages/shared/src/components/modals/NewSourceModal.tsx:186]: What are we using the ID for?

File: packages/shared/src/components/modals/NewSourceModal.tsx [hunk 2]
```diff
@@ -262,26 +258,41 @@ export default function NewSourceModal(props: ModalProps): ReactElement {
...
       </Modal.Body>
-      {(!!feeds?.length || isScraping) && (
-        <Modal.Footer>
-          {isScraping && (
-            <div className="flex justify-center w-full">
-              <Loader invertColor />
-            </div>
-          )}
-          {!!feeds?.length && (
+      <Modal.Footer>
```
Comment (rebelchris) [inline packages/shared/src/components/modals/NewSourceModal.tsx:261]: Should this not have justify between for the one with two btutons? (I do see you used mr-auto)

**Review by rebelchris — APPROVED** [review]
Nothing blocking, looks great. 🥳
Do have some suggestions we can consider.

File: packages/shared/src/components/modals/NewSourceModal.tsx [hunk 3]
```diff
@@ -195,13 +196,8 @@ export default function NewSourceModal(props: ModalProps): ReactElement {
...
             aria-describedby={scrapeError && 'new-source-field-desc'}
             valueChanged={onUrlChanged}
-            fieldType="secondary"
-            rightButtonProps={{
-              disabled: !enableSubmission,
-              type: 'submit',
-              'aria-label': 'Search feeds',
-              buttonSize: 'small',
-            }}
+            fieldType="primary"
```
Comment (sshanzel) [inline packages/shared/src/components/modals/NewSourceModal.tsx:199]: Is it part of the new design to turn this into a primary?

File: packages/shared/src/components/fields/SearchField.tsx [hunk 4]
```diff
@@ -102,31 +103,33 @@ export const SearchField = forwardRef(function SearchField(
...
-          className="mr-2 text-2xl icon"
-          style={{
-            color:
-              focused || hasInput
-                ? 'var(--theme-label-primary)'
-                : 'var(--field-placeholder-color)',
-          }}
-        />
-      )}
+      {!!showIcon &&
```
Comment (sshanzel) [inline packages/shared/src/components/fields/SearchField.tsx:106]: The `showIcon` should not be needed anymore, I think it was forgotten to be removed but as part of being a secondary type (in the search field guideline), icon should be displayed.

**Review by sshanzel — COMMENTED** [review]
Amazing work on refactoring this! 🚢 
Just had a few non-blocking comments/questions around the search field guideline.

File: packages/shared/src/components/modals/NewSourceModal.tsx (hunk 2 above)
Comment (el) [reply packages/shared/src/components/modals/NewSourceModal.tsx:261]: Dont they have the same outcome?

File: packages/shared/src/components/modals/NewSourceModal.tsx (hunk 3 above)
Comment (el) [reply packages/shared/src/components/modals/NewSourceModal.tsx:199]: yeap https://www.sketch.com/s/2867ec63-0966-4647-b1d7-b6fe696ba251/a/m1913YJ

File: packages/shared/src/components/fields/SearchField.tsx (hunk 4 above)
Comment (el) [reply packages/shared/src/components/fields/SearchField.tsx:106]: New design does not have an icon
https://www.sketch.com/s/2867ec63-0966-4647-b1d7-b6fe696ba251/a/m1913YJ

File: packages/shared/src/components/modals/NewSourceModal.tsx (hunk 1 above)
Comment (el) [reply packages/shared/src/components/modals/NewSourceModal.tsx:186]: Here:
https://github.com/dailydotdev/apps/pull/1404/files#diff-2b63e881f1f68e2233cfd08a081ff1d03677b0b621b65ad9e8dc7ccc4777369eR264

**Pull request merged**